src/sfctl/tests/request_generation_test.py

```diff
@@ -358,6 +358,37 @@ def paths_generation_helper(self):  # pylint: disable=too-many-statements
             '{"CreateFabricDump":"True", "NodeInstanceId":"ID"}',
             validate_flat_dictionary)
 
+        # container commands
+        self.validate_command( # get container logs
+            'sfctl container invoke-api --node-name Node01 --application-id samples/winnodejs '
+            '--service-manifest-name NodeServicePackage --code-package-name NodeService.Code '
+            '--code-package-instance-id 131668159770315380 --container-api-uri-path "/containers/{id}/logs?stdout=true&stderr=true"',
+            'POST',
+            '/Nodes/Node01/$/GetApplications/samples/winnodejs/$/GetCodePackages/$/ContainerApi',
+            ['api-version=6.2', 'ServiceManifestName=NodeServicePackage', 'CodePackageName=NodeService.Code', 'CodePackageInstanceId=131668159770315380', 'timeout=60'],
+            ('{"UriPath": "/containers/{id}/logs?stdout=true&stderr=true"}'),
+            validate_flat_dictionary)
+        self.validate_command( # get container logs
+            'sfctl container logs --node-name Node01 --application-id samples/winnodejs '
+            '--service-manifest-name NodeServicePackage --code-package-name NodeService.Code '
+            '--code-package-instance-id 131668159770315380',
+            'POST',
+            '/Nodes/Node01/$/GetApplications/samples/winnodejs/$/GetCodePackages/$/ContainerApi',
+            ['api-version=6.2', 'ServiceManifestName=NodeServicePackage', 'CodePackageName=NodeService.Code', 'CodePackageInstanceId=131668159770315380', 'timeout=60'],
+            ('{"UriPath": "/containers/{id}/logs?stdout=true&stderr=true"}'),
+            validate_flat_dictionary)
+        # self.validate_command( # update container
+        #     'sfctl container invoke-api --node-name N0020 --application-id nodejs1 --service-manifest-name NodeOnSF '
+        #     '--code-package-name Code --code-package-instance-id 131673596679688285 --container-api-uri-path "/containers/{id}/update"'
+        #     ' --container-api-http-verb=POST --container-api-body "{ \"CpuShares\": 512}"',
+        #     'POST',
+        #     '/Nodes/N0020/$/GetApplications/nodejs1/$/GetCodePackages/$/ContainerApi',
+        #     ['api-version=6.2', 'ServiceManifestName=NodeOnSF', 'CodePackageName=Code', 'CodePackageInstanceId=131673596679688285', 'timeout=60'],
+        #     ('{"UriPath": "/containers/{id}/update", '
+        #      '"HttpVerb": "POST", '
+        #      '"Body": "{ \"CpuShares\": 512}"}'),
```

Comment:
I think the test is failing because of this. Because of how validate_flat_dictionary works. It does a json.loads, and somehow that always gets confused. If you've verified by hand that passing in a json works,  you can change the "Body" input to not be a json and test it that way. And just pls make a note in the test about this.